script.js

```diff
@@ -1,12 +1,461 @@
+}
+
+function changeEpisode(e) {
+  let selectedOption = e.target.value;
+  let getResultsOnPage = document.getElementsByClassName("wrapper");                             //I declared this variable again (there is one in level 200), did not want to have anything in the global scope, so not sure if it is best practice?
+  let searchResult = document.querySelector(".mySearchResult");
+  let allResultsOnPage = Array.from(getResultsOnPage);
+
+  allResultsOnPage.forEach(result => {
+    if (result.innerText.includes(selectedOption)) {
+      result.classList.remove("hidden");
+    } else {
+      result.classList.add("hidden");
+    }
+
+    searchResult.innerText = `Displaying 1/${getResultsOnPage.length} results`;
+
+    if (selectedOption === "All Results") {
+      result.classList.remove("hidden");
+      searchResult.innerText = `Displaying ${getResultsOnPage.length} results`;
+    }
+
+  });
+}
+
+//Level 400
+
+function addShowsSelector() {
+  let selectAShow = document.createElement("select");
+  selectAShow.setAttribute("id", "shows");
+
+  let chooseEpisodeAndShow = document.createElement("label");
+  chooseEpisodeAndShow.innerText = "Choose a show";
+  chooseEpisodeAndShow.classList.add("chooseLabel");
+
+  let pageNavigation = document.querySelector("nav");
+  pageNavigation.prepend(chooseEpisodeAndShow, selectAShow);
+
+  let showsList = getSortedShowsList();
+
+  for (let showIndex = 0; showIndex < showsList.length; showIndex++) {
+    let showOption = document.createElement("option");
+    selectAShow.appendChild(showOption);
+
+    showOption.innerText = showsList[showIndex].name;
+    showOption.setAttribute("value", `${showsList[showIndex].id}`);
+
+    if (showsList[showIndex].id === showID) {                                                   // This condition is added so when an option is selected from the "shows selector", the default option will be what the user selected and not the very first option "24";
+      showOption.setAttribute("selected", `${showsList[showIndex]}`);
+      showID = 0;
+    }
+  }
+
+  selectAShow.addEventListener("change", changeShow);
+}
+
+function changeShow(e) {
+  showID = +e.target.value;
+
+  removeCurrentPage();
+
+  getAllEpisodes(showID);
+}
+
+function removeCurrentPage() {
+  let rootElem = document.getElementById("root");
+  rootElem.innerHTML = "";
+
+  let navigationWrapper = document.querySelector(".navigationWrapper");
+  navigationWrapper.innerHTML = "";
+}
+
+
+//Level 500 part 1 and 2
+
+function getSortedShowsList() {
+  let allShows = getAllShows();
+
+  let sortedShowsList = allShows.sort(function compare(showA, showB) {
+    if (showA.name.toUpperCase() < showB.name.toLocaleUpperCase()) { return -1 }
+    if (showA.name.toUpperCase() > showB.name.toUpperCase()) { return 1 }
+    return 0;
+  })
+
+  return sortedShowsList;
+}
+
+function makePageForShowsList() {
+  let showsList = getSortedShowsList()
+
+  let rootElem = document.getElementById("root");
```

Comment:
I have noticed you used this variable couple of times. I think it would be better if you want to use a variable a few times, put it in the global scope.

Reply:
@DanGhanbari Thanks Daneil for this feedback, you are right, I can have it in the global scope.